### PR TITLE
feat: add fuzzy text search support

### LINKED ISSUE #4: [Bug]: Search work without file extension, but with it's not
^ Title
---
For example i have file it's name is `some_random_word_yes_it_is.md` in `/home/User/foo/bar/`
```rust
use rust_search::Search;

fn main(){
    let search = Search::new(
        "/home/User/foo/bar/",
        Some("yes"),
        None,
        None);

    for path in search {
        println!("{}", path);
    }
}
```
This will work, but when i add the file extinction it's not, for example

```rust
use rust_search::Search;

fn main(){
    let search = Search::new(
        "/home/User/foo/bar/",
        Some("yes"),
        Some(".md"), // also "md"
        None);

    for path in search {
        println!("{}", path);
    }
}
```

### PR DESCRIPTION
**Pull request by ParthJadhav**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -60,8 +60,9 @@ impl Search {
         search_input: Option<&str>,
         file_type: Option<&str>,
         depth: Option<usize>,
+        strict: Option<bool>,
```
Comment (TheAwiteb) [inline src/lib.rs:63]: This should be `bool` not `Option<bool>`

**Review by TheAwiteb — CHANGES_REQUESTED** [review]

**Comment by TheAwiteb** [comment]
Hey @ParthJadhav, don't release this PR, because it and #6 have a BC, so release both in one major version

**Comment by ParthJadhav** [comment]
Alright cool.

**Comment by TheAwiteb** [comment]
@ParthJadhav there is no strict when search without extinction, i think it's better not to have strict with the extension too, do you think?

**Comment by ParthJadhav** [comment]
@TheAwiteb added this feature to below PR. 
https://github.com/ParthJadhav/rust_search/pull/9

> @ParthJadhav there is no strict when search without extinction, i think it's better not to have strict with the extension too, do you think?

Yes, It defaults to be `not strict` when passed `None`

**Pull request closed without merging**